pkg/lottery/contract_test.go

```diff
@@ -0,0 +1,84 @@
+package lottery_test
+
+type TestContract struct {
+	config func() (startBlock, blocksPerRound, BlocksPerPhase uint64)
+	// accessors for random sources available in claim phase
+	randomSeedFor func(int) ([]byte, error)
+	// function that tells the node whether to claim
+	isWinner func(overlay, truthSel, winnerSel []byte) (slashed, winner bool, err error)
+	// function that tells the node whether to commit (start sampler)
+	isPlaying func(overlay []byte, sd uint8, source []byte) (bool, error)
+	// transactional calls
+	commit func(rc []byte) error
+	reveal func(sd uint8, rc, key []byte) error
+	claim  func() error
+}
+
+func (c *TestContract) WithConfig(f func() (startBlock, blocksPerRound, BlocksPerPhase uint64)) *TestContract {
```

Comment:
Usually, we use the `Options` pattern in these cases. So the `New` function would take these options and initialize the components required.

Reply:
ok I rewrite this then